src/stability_sdk/animation.py

```diff
@@ -219,6 +220,17 @@ def save_settings(self, filename: str):
             save_dict['negative_prompt_weight'] = self.negative_prompt_weight
             json.dump(save_dict, f, ensure_ascii=False, indent=4)
 
+    def prepare_init_image(self, fpath=None):
+        if fpath is None:
+            #fpath = self.args.get('init_image')
+            fpath =  self.args.init_image
+        if not fpath:
+            return
+        img = Image.open(fpath)
+        self.args.height, self.args.width = img.size[0], img.size[1]
```

Comment:
that's the path I was originally going, but I like the ergonomics of this better. i consider most of this temporary scaffolding anyway: I'm already working on a modification of this that let's the user specify per-keyframe init images. let's put a pin in this and discuss ergonomics and default behaviors when we're closer to release. whatever we choose, I agree that video and image init should probably behave similarly. I'm not convinced forcing the user to specify dimensions to match their init is a good user experience, but I also want to separate "smart defaults" tooling from making functionality available at all :P

Reply:
hmm are you sure? other tools like dreamstudio, deforum, etc modify input image to fix the target resolution rather than change the target resolution to match init image. if it is kept this way it needs to at least modify the dimensions so they are multiple of 64 and stretch or crop to fit that.